src/main/java/graphql/validation/AbstractRule.java

```diff
@@ -51,8 +53,16 @@ public void setValidationUtil(ValidationUtil validationUtil) {
         this.validationUtil = validationUtil;
     }
 
-    public void addError(ValidationError error) {
```

Comment:
Path is tracked by the traversal context and these little factory methods make errors that add the path